tests/test_model.py

```diff
@@ -1267,6 +1268,33 @@ def test_assign_genres_from_weights(self):
         after = sorted((x.genre.name, x.affinity) for x in work.work_genres)
         eq_([(u'Romance', 0.25), (u'Science Fiction', 0.75)], after)
 
+    def test_classifications_with_genre(self):
+        work = self._work()
+        identifier = work.primary_edition.primary_identifier
+        genres = self._db.query(Genre).all()
+        subject1 = self._subject(type="type1", identifier="subject1", genre=genres[0])
+        subject2 = self._subject(type="type2", identifier="subject2", genre=genres[1])
+        subject3 = self._subject(type="type2", identifier="subject3", genre=None)
+        source = DataSource.lookup(self._db, DataSource.AXIS_360)        
+        classification1 = self._classification(
+            identifier=identifier, subject=subject1, 
+            data_source=source, weight=1)
+        classification2 = self._classification(
+            identifier=identifier, subject=subject2, 
+            data_source=source, weight=2)
+                
+        results = work.classifications_with_genre().all()
+        
```

Comment:
Are these two items `classification1` and `classification2`? I think they might be. If so, you can check them for equality instead of checking that all the attributes match:

```
eq_([classification1, classification2], work.classifications_with_genre().all())
```

However you'll also need to make sure to sort the query, since sometimes databases return things in a weird order. Something like this should work:

```
classification_q = work.classifications_with_genre().order_by(Classification.id)
eq_([classification1, classification2], classification_q.all())
```